lib/ansible/modules/system/setup.py

```diff
@@ -103,6 +103,15 @@
 # Collect only facts returned by facter.
 # ansible all -m setup -a 'gather_subset=!all,!any,facter'
 
+- name: Collect default facts and add them to variable setupvar
+  setup:
+  register: setupvar
+
+- name: Store facts indexed by hostname in /tmp/facts
+  copy:
+    content: '{{ setupvar }}'
+    dest: /tmp/facts/{{ ansible_hostname }}
```

Comment:
Why can't you just utilize `hostvars[inventory_hostname]` instead of running `setup`+`register` and using that var?

Using `register` with modules that return `ansible_facts` should be considered an anti-pattern, and not something that is recommended.